src/components/Fetcher/index.js

```diff
@@ -0,0 +1,133 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import Spinner from 'react-spinkit'
+import { connect } from 'react-redux'
+
+import LANGUAGE_FETCHER from 'endpoints/language'
+import PAGE_ENDPOINT from 'endpoints/page'
+import DISCLAIMER_FETCHER from 'endpoints/disclaimer'
+import LOCATION_FETCHER from 'endpoints/location'
+
+import Error from 'components/Error'
+
+import style from './style.css'
+
+function createFetcher (endpoint,
+                        createUrlOptions, createUrlTransformOptions,
+                        mapStateToProps = () => { return {} },
+                        shouldUpdate = () => false) {
+  let stateName = endpoint.name
+  let payloadName = stateName + 'Payload'
+
+  function mapStateToFetcherProps (state) {
+    let props = mapStateToProps(state)
+
+    props[payloadName] = state[stateName]
+    return props
+  }
+
+  let Fetcher = class extends React.PureComponent {
```

Comment:
The props of a Fetcher are generally more simple, they are just params for the http query. But I'll change it back to Component. There's no reason to optimize if we have no probs :D